ehb_service/apps/api/views/externalrecord.py

```diff
@@ -1,25 +1,24 @@
 import json
 import logging
 
-from restlib2.resources import Resource
-from restlib2.http import codes
-from django.http import HttpResponse
 from django.db.models import Q
 from core.forms import ExternalRecordForm, ExternalRecordRelationForm
 from constants import ErrorConstants
 from api.helpers import FormHelpers
 
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.decorators import permission_classes
+from rest_framework import permissions
+
 from core.models.identities import ExternalRecord, \
     ExternalRecordRelation, ExternalRecordLabel, Subject, ExternalSystem
 
 log = logging.getLogger(__name__)
 
-
-class ExternalRecordQuery(Resource):
-
-    supported_accept_types = ['application/json']
-
-    model = 'core.models.identities.ExternalRecord'
+@permission_classes((permissions.AllowAny,))
```

Comment:
Removed the @permissionsAllowAny header from all the classes. Created a api/authentication.py file to override Django authentication so that we can continue authenticating using the api tokens that we've created instead of switching over to Django-generated auth-tokens. 803fea7fede60b0fad4d6f72f324d7be87822a99